### PR TITLE
Migrate PR tests over to GH Runner

### PR DESCRIPTION
**Pull request by ehildenb**
This migrates the Jenkins job which runs tests on PRs over to GH Runner.

- [x] **TODO**: We need to update the required status checks prior to merge.

### CODE REVIEW

**Comment by ehildenb** [comment]
@bbyalcinkaya and @hjorthjort I've checked that it looks like we're running all the saem tests as Jenkins using this new setup, but it would be good for you to double-check.

DO NOT MERGE until I've updated the required status checks.

File: .github/workflows/test-pr.yml [hunk 1]
```diff
@@ -41,10 +41,83 @@ jobs:
...
+      - name: 'Set up Docker'
+        uses: ./.github/actions/with-docker
+        with:
+          container-name: kwasm-ci-conformance-${{ github.sha }}
+      - name: 'Build LLVM Backend'
+        run: docker exec -u user kwasm-ci-conformance-${GITHUB_SHA} make -j2 build-llvm
+      - name: 'Simple tests'
+        run: docker exec -u user kwasm-ci-conformance-${GITHUB_SHA} make -j6 test-simple
+      - name: 'Conformance run'
+        run: docker exec -u user kwasm-ci-conformance-${GITHUB_SHA} make -j6 test-conformance-supported
```
Comment (F-WRunTime) [inline .github/workflows/test-pr.yml:96]: In the Jenkisn run this also has TEST_CONCRETE_BACKEND
`make TEST_CONCRETE_BACKEND=llvm test-conformance-supported -j4`

File: .github/workflows/test-pr.yml [hunk 2]
```diff
@@ -41,10 +41,83 @@ jobs:
...
+        with:
+          submodules: recursive
+      - name: 'Set up Docker'
+        uses: ./.github/actions/with-docker
+        with:
+          container-name: kwasm-ci-conformance-${{ github.sha }}
+      - name: 'Build LLVM Backend'
+        run: docker exec -u user kwasm-ci-conformance-${GITHUB_SHA} make -j2 build-llvm
+      - name: 'Simple tests'
+        run: docker exec -u user kwasm-ci-conformance-${GITHUB_SHA} make -j6 test-simple
```
Comment (F-WRunTime) [inline .github/workflows/test-pr.yml:94]: Also appears to be missing TEST_CONCRETE
`make TEST_CONCRETE_BACKEND=llvm test-simple -j4`

File: .github/workflows/test-pr.yml [hunk 3]
```diff
@@ -41,10 +41,83 @@ jobs:
...
+        with:
+          submodules: recursive
+      - name: 'Set up Docker'
+        uses: ./.github/actions/with-docker
+        with:
+          container-name: kwasm-ci-parse-${{ github.sha }}
+      - name: 'Build LLVM Backend and pykwasm'
+        run: docker exec -u user kwasm-ci-parse-${GITHUB_SHA} make -j2 build-llvm pykwasm-poetry-install
+      - name: 'Binary parse'
+        run: docker exec -u user kwasm-ci-parse-${GITHUB_SHA} make -j6 test-binary-parser
```
[the remark this answers is not in the document]
Comment (F-WRunTime) [reply .github/workflows/test-pr.yml:69]: ignore -j here just copy pasting from the jenkin logs

**Comment by F-WRunTime** [comment]
Besides the above. I see full replication of existing tests from Jenkins to GH

**Review by F-WRunTime — APPROVED** [review]

File: .github/workflows/test-pr.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (ehildenb) [reply .github/workflows/test-pr.yml]: Done.

**Pull request merged**